include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,619 @@
+ * The optimal dictionaries are application specific
+ * The dictionary may be fixed or may be chosen among a certain number of
+ * predefined dictionaries, according to the kind of input data
+ *
+ * Refer RFC 1950
+ */
+typedef struct odp_comp_dict_t {
+	/** pointer to byte array */
+	uint8_t *buf;
+	/** length of the dictionary */
+	uint32_t len;
+} odp_comp_dict_t;
+
+/**
+ * Compression algorithm capabilities
+ */
+typedef struct odp_comp_alg_capability_t {
+
+	/** Maximum length of dictionary supported for this algorithm
+	 * Value of zero indicates dictionary is not supported.
+	 */
+	uint32_t dict_len;
+
+	/** Maximum compression level supported by implementation of this
+	 * algorithm. Indicates number of compression levels supported by
+	 * implementation. Valid range from (1 ... max_level)
+	 */
+	uint32_t max_level;
+
+	/** Supported hash algorithms */
+	odp_comp_hash_algos_t hash_algo;
+} odp_comp_alg_capability_t;
+
+/**
+ * Compression Huffman type. Used by DEFLATE algorithm
+ */
+typedef enum odp_comp_huffman_code {
+	/** Fixed Huffman code */
+	ODP_COMP_HUFFMAN_FIXED,
+
+	/** Dynamic Huffman code */
+	ODP_COMP_HUFFMAN_DYNAMIC,
+
+	/** Default huffman code selected by implementation */
+	ODP_COMP_HUFFMAN_DEFAULT,
+} odp_comp_huffman_code_t;
+
+/**
+ * Compression DEFLATEe algorithm parameters.
+ * Also initialized by other deflate based algorithms , ex. ZLIB
+ */
+typedef struct odp_comp_alg_def_param {
+	/**
+	 * Compression level
+	 *
+	 * Valid range is integer between (0 ... max_level)
+	 * level supported by the implementation.
+	 *
+	 * where,
+	 * 0 - implemention default
+	 *
+ 	 * 1 - fastest compression i.e. output produced at
+ 	 * best possible speed at the expense of compression quality
+	 *
+	 * max_level - High quality compression
+	 *
+	 * @see 'max_level' in odp_comp_alg_capability_t
+	 */
+	uint32_t comp_level;
+
+	/** huffman code to use */
+	odp_comp_huffman_code_t comp_code;
+} odp_comp_alg_def_param_t;
+
+/**
+ * Compression algorithm specific parameters
+ */
+typedef union odp_comp_alg_param_t {
+	/** deflate parameter */
+	odp_comp_alg_def_param_t deflate;
+
+	/**
+	 * Struct for defining zlib algorithm parameters
+	 */
+	struct comp_alg_zlib_param {
+		/** deflate algo params */
+		odp_comp_alg_def_param_t def;
+	} zlib; /** zlib parameter */
+} odp_comp_alg_param_t;
+
+ /**
+ * Compression session creation parameters
+ */
+typedef struct odp_comp_session_param_t {
+	/** Compression operation type Compress vs Decompress */
+	odp_comp_op_t op;
+
+	/** Compression operation mode
+	 *
+	 * Operation mode Synchronous vs Asynchronous
+	 *
+	 * @see odp_comp_op(), odp_comp_op_enq()
+	 */
+	odp_comp_op_mode_t mode;
+
+	/** Compression algorithm
+	 *
+	 *  @see odp_comp_capability()
+	 */
+	odp_comp_alg_t comp_algo;
+
+	/** Hash algorithm
+	 *
+	 *  @see odp_comp_alg_capability()
+	 */
+	odp_comp_hash_alg_t hash_algo;
+
+	/** parameters specific to compression */
+	odp_comp_alg_param_t algo_param;
```

Comment:
Field name should match the type name. Field/type names are easier to remember / search when those match.
algo_param ==> alg_param

Reply:
Will update in next version